sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs

```diff
@@ -823,19 +953,20 @@ public async Task RetriesOnRetriableStatusCode([Values(404, 410, 500)] int statu
             var errorMessage = "Some error happened";
             var mockTransport = new MockTransport(request =>
             {
-                retryCount++;
+                tryCount++;
                 return CreateErrorMockResponse(status, errorMessage);
             });
             var options = new TokenCredentialOptions { Transport = mockTransport };
             options.Retry.MaxDelay = TimeSpan.Zero;
-            var pipeline = CredentialPipeline.GetInstance(options);
+            var pipeline = CredentialPipeline.GetInstance(options, true);
 
-            ManagedIdentityCredential credential = InstrumentClient(new ManagedIdentityCredential("mock-client-id", pipeline));
+            ManagedIdentityCredential credential = InstrumentClient(new ManagedIdentityCredential(
+                new ManagedIdentityClient(
+                    new ManagedIdentityClientOptions() { Pipeline = pipeline, ClientId = "mock-client-id", IsForceRefreshEnabled = true, Options = options })));
 
             var ex = Assert.ThrowsAsync<AuthenticationFailedException>(async () => await credential.GetTokenAsync(new TokenRequestContext(MockScopes.Default)));
-            Assert.IsInstanceOf(typeof(RequestFailedException), ex.InnerException);
             Assert.That(ex.Message, Does.Contain(errorMessage));
-            Assert.That(retryCount, Is.EqualTo(4));
+            Assert.That(tryCount, Is.EqualTo(6));
```

Comment:
With this change, we are now sending 2 more requests?

Reply:
I believe the new value is actually correct. We are [retrying 5 times](https://github.com/Azure/azure-sdk-for-net/blob/938d042748a8d75ef5fa725d3fc9aa261a56f041/sdk/identity/Azure.Identity/src/CredentialPipeline.cs#L48) with one initial request.